src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -362,10 +362,28 @@ private function parseDefinition(\DOMElement $service, string $file, array $defa
             $definition->setBindings($bindings);
         }
 
-        if ($value = $service->getAttribute('decorates')) {
+        if ($decorates = $service->getAttribute('decorates')) {
+            $onDecoratedInvalid = $service->getAttribute('decoration-on-decorated-invalid') ?: 'exception';
+            if (!\in_array($onDecoratedInvalid, ['exception', 'ignored', 'nulled'])) {
+                throw new InvalidArgumentException(sprintf('Invalid attribute "decoration-on-decorated-invalid" defined for "%s" in "%s"', (string) $service->getAttribute('id'), $file));
+            }
+
             $renameId = $service->hasAttribute('decoration-inner-name') ? $service->getAttribute('decoration-inner-name') : null;
             $priority = $service->hasAttribute('decoration-priority') ? $service->getAttribute('decoration-priority') : 0;
-            $definition->setDecoratedService($value, $renameId, $priority);
+
+            switch ($onDecoratedInvalid) {
```

Comment:
the switch is missing a "default" case where we would throw - btw, should the switch be replaced by if/elseif/else?

Reply:
oh, there is a check before. better move the check in the "else" case, that's ease understanding the code.
Also, the exception message should tell about the possible values, and the wrong one.